### PR TITLE
CURA-5334 Add extruder definitions for all single-extrusion machines

### PR DESCRIPTION
**Pull request by LipuFei**
All modified machines:
 - 101Hero
 - 3dator
 - abax_pri3
 - abax_pri5
 - abax_titan
 - alya3dp
 - anycubic_i3_mega
 - bfb
 - bq_hephestos
 - bq_hephestos_2
 - bq_hephestos_xl
 - bq_witbox
 - bq_witbox_2
 - creality_cr10
 - creality_cr10s4
 - creality_cr10s5
 - dagoma_discoeasy200
 - dagoma_neva
 - delta_go
 - deltabot
 - deltacomb
 - easyarts_ares
 - fabtotum
 - folgertech_FT-5
 - gmax15plus
 - grr_neo
 - imade3d_jellybox
 - innovo_inventor
 - julia
 - kemiq_q2_beta
 - kemiq_q2_gama
 - kossel_mini
 - kossel_pro
 - kupido
 - makeR_pegasus
 - makeR_prusa_tairona_i3
 - maker_starter
 - makerbotreplicator
 - malyan_m180
 - malyan_m200
 - mankati_fullscale_xt_plus
 - mendel90
 - monoprice_select_mini_v1
 - monoprice_select_mini_v2
 - peopoly_moai
 - printrbot_play
 - printrbot_play_heated
 - printrbot_simple
 - printrbot_simple_extended
 - printrbot_simple_makers_kit
 - prusa_i3
 - prusa_i3_mk2
 - prusa_i3_xl
 - raise3D_N2_plus_single
 - raise3D_N2_single
 - renkforce_rf100
 - rigid3d
 - rigid3d_3rdgen
 - rigid3d_hobby
 - rigid3d_mucit
 - rigid3d_zero
 - rigid3d_zero2
 - rigidbot
 - rigidbot_big
 - robo_3d_r1
 - seemecnc_artemis
 - seemecnc_v32
 - tam
 - tevo_blackwidow
 - tevo_tarantula
 - ubuild-3d_mr_bot_280
 - ultimaker2
 - ultimaker2_extended
 - ultimaker2_extended_plus
 - ultimaker2_go
 - ultimaker2_plus
 - ultimaker_original
 - ultimaker_original_plus
 - uniqbot_one
 - vertex_delta_k8800
 - vertex_k8400
 - zone3d_printer
 - zyyx_agile

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
While it adds some files to maintain (which probably will be separate in the near future if we move machines to the toolbox as well), it seems it's a more unified solution for single & dual extrusion.

File: cura/Settings/ExtruderManager.py [hunk 1]
```diff
@@ -393,73 +393,21 @@ def _addCurrentMachineExtruders(self) -> None:
...
-        # the extruders at this point.
-        for key in keys_to_copy:
-            if global_stack.definitionChanges.hasProperty(key, "value"):
-                global_stack.definitionChanges.removeInstance(key, postpone_emit = True)
+    # After 3.4, all single-extrusion machines have their own extruder definition files instead of reusing
+    # "fdmextruder". We need to check a machine here so its extruder definition is correct according to this.
+    def _fixSingleExtrusionMachineExtruderDefinition(self, global_stack):
+        expected_extruder_definition_0_id = global_stack.getMetaDataEntry("machine_extruder_trains")["0"]
+        extruder_stack_0 = global_stack.extruders["0"]
+        if extruder_stack_0.definition.getId() != expected_extruder_definition_0_id:
```
[the remark this answers is not in the document]
Comment (jackha) [reply cura/Settings/ExtruderManager.py]: Done in 955ed00c39f69aef533a94b4c29dd611bb8a05cb

File: resources/definitions/ord.def.json [hunk 2]
```diff
@@ -23,7 +23,6 @@
         "prime_tower_size": { "default_value": 7.0710678118654755 },
         "infill_sparse_density": { "default_value": 15 },
         "speed_travel": { "default_value": 150 },
-        "material_diameter": { "default_value": 1.75 },
```
Comment (diegopradogesto) [inline resources/definitions/ord.def.json]: Removed from here, but not moved to the extruder definition.

File: resources/definitions/punchtec_connect_xl.def.json [hunk 3]
```diff
@@ -21,7 +21,6 @@
         "prime_tower_size": { "default_value": 8.660254037844387 },
         "speed_wall_x": { "default_value": 40 },
         "speed_wall_0": { "default_value": 40 },
-        "material_diameter": { "default_value": 1.75 },
```
Comment (diegopradogesto) [inline resources/definitions/punchtec_connect_xl.def.json]: Removed from here, but not moved to the extruder definition.

File: resources/definitions/raise3D_N2_dual.def.json [hunk 4]
```diff
@@ -54,9 +54,6 @@
         "machine_min_cool_heat_time_window": {
             "default_value": 3600
         },
-        "material_diameter": {
-            "default_value": 1.75
-        },
```
Comment (diegopradogesto) [inline resources/definitions/raise3D_N2_dual.def.json]: Removed from here, but not moved to the extruder definition.

File: resources/definitions/raise3D_N2_plus_dual.def.json [hunk 5]
```diff
@@ -54,9 +54,6 @@
         "machine_min_cool_heat_time_window": {
             "default_value": 3600
         },
-        "material_diameter": {
-            "default_value": 1.75
-        },
```
Comment (diegopradogesto) [inline resources/definitions/raise3D_N2_plus_dual.def.json]: Removed from here, but not moved to the extruder definition.

File: resources/definitions/vertex_k8400_dual.def.json [hunk 6]
```diff
@@ -46,9 +46,6 @@
         "machine_nozzle_size": {
             "default_value": 0.35
         },
-        "material_diameter": {
-            "default_value": 1.75
-        },
```
Comment (diegopradogesto) [inline resources/definitions/vertex_k8400_dual.def.json]: Removed from here, but not moved to the extruder definition.

File: resources/definitions/builder_premium_large.def.json [hunk 7]
```diff
@@ -26,7 +26,6 @@
         "machine_width": { "default_value": 215 },
         "machine_height": { "default_value": 600 },
         "machine_depth": { "default_value": 205 },
-        "material_diameter": { "default_value": 1.75 },
```
[the remark this answers is not in the document]
Comment (LipuFei) [reply resources/definitions/builder_premium_large.def.json]: They are already there in commit: aba3795d0b51b3ca8fa5307bd380a0a8fade6aff
@jackha just cleaned them up from the machine definitions.

[the remark this answers is not in the document]
Comment (diegopradogesto) [reply resources/definitions/builder_premium_large.def.json]: This is not yet done. What I meant is that for all those printers that already had extruder files, you deleted the `material_diameter` but not moved to those extruder files.

File: resources/definitions/malyan_m180.def.json [hunk 8]
```diff
@@ -62,11 +66,6 @@
         },
         "machine_end_gcode": {
             "default_value": "G92 Z0\nG1 Z10 F400\nM18\nM109 S0 T0\nM104 S0 T0\nM73 P100 (end  build progress)\nG162 X Y F3000\nM18"
-        },
-        "material_diameter": {
-            "default_value": 1.75,
-            "minimum_value_warning": "1.5",
-            "maximum_value_warning": "2.0"
```
[the remark this answers is not in the document]
Comment (jackha) [reply resources/definitions/malyan_m180.def.json]: We think it can be left out.

**Comment by jackha** [comment]
All 1.75 extruders now have material_diameter filled in.

**Pull request merged**